indra/newview/llpuppetmotion.cpp

```diff
@@ -343,6 +343,59 @@ void LLPuppetMotion::rememberPosedJoint(S16 joint_id, LLPointer<LLJointState> jo
     }
 }
 
+void LLPuppetMotion::reportRootRelativePosition(const S16 joint_id, const S32 request_id)
+{
+    //Get current and root joints.
+    LLPointer<LLJointState> joint_state = mJointStates[joint_id];
+    LLPointer<LLJointState> root_state = mJointStates[0];
+        
+    if (!joint_state || !root_state) return;
+    std::string name = joint_state->getJoint()->getName();
+
+    LLVector3 rpos = root_state->getJoint()->getWorldPosition();
+    LLVector3 jpos = joint_state->getJoint()->getWorldPosition();
+    LLVector3 jend = joint_state->getJoint()->getEnd();
+    
+    jend.rotVec( joint_state->getJoint()->getWorldRotation());
+    jpos = jpos + jend;
+    
+    LLVector3 opos = jpos - rpos; //Vector from root to joint end
+    LLQuaternion rrot = root_state->getJoint()->getWorldRotation();
+    
+    opos.rotVec(~rrot); //remove root rotation.
+    
+    F32 span = mArmSpan * 0.5f;
+    LLVector3 opos2 = opos / span; //scale result for armspan.
+    
+    rrot = joint_state->getJoint()->getRotation();
+    LLVector3 reuler = rrot.getEulerAngles() * RAD_TO_DEG;
+    
+    
+    //Has joint ID, 1M scale, armscale.
+    LL_INFOS("Puppet") << " PostIK report: " << name << " (" <<
+        joint_id << ") 1M scale position: (" <<
+        opos.mV[0] << ", " << opos.mV[1] << ", " << opos.mV[2] <<
```

Comment:
`LLVector3` should have its own `std::ostream& operator<<(std::ostream&, const LLVector3&)` overload.